src/test/java/seedu/guest/commons/util/StringUtilTest.java

```diff
@@ -123,6 +124,157 @@ public void containsWordIgnoreCase_validInputs_correctResult() {
         assertTrue(StringUtil.containsWordIgnoreCase("AAA bBb ccc  bbb", "bbB"));
     }
 
+    //---------------- Tests for containsDateIgnoreHyphenIgnoreSpace --------------------------------------
+
+    /*
+     * Invalid equivalence partitions for date: null, empty, multiple dates
+     * Invalid equivalence partitions for sentence: null
+     * The four test cases below test one invalid input at a time.
+     */
+
+    @Test
+    public void containsDateIgnoreHyphenIgnoreSpace_nullDate_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> StringUtil.containsDateIgnoreHyphenIgnoreSpace(
+                "13/03/22", null));
+    }
+
+    @Test
+    public void containsDateIgnoreHyphenIgnoreSpace_emptyDate_returnsFalse() {
+        assertEquals(false, StringUtil.containsDateIgnoreHyphenIgnoreSpace("13/03/22", "  "));
+    }
+
+    @Test
+    public void containsDateIgnoreHyphenIgnoreSpace_multipleDates_throwsIllegalArgumentException() {
+        assertThrows(IllegalArgumentException.class, "Date parameter should be a single date", ()
+                -> StringUtil.containsDateIgnoreHyphenIgnoreSpace("13/03/22", "13/03/22 14/03/22"));
+    }
+
+    @Test
+    public void containsDateIgnoreHyphenIgnoreSpace_nullSentence_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> StringUtil.containsDateIgnoreHyphenIgnoreSpace(
+                null, "13/03/22"));
+    }
+
+    /*
+     * Valid equivalence partitions for date:
+     *   - valid date
+     *   - date with leading/trailing spaces
```

Comment:
Should we also have a test case to check if date with leading and trailing spaces would also return true?

Reply:
Agreed.